odoo/models.py

```diff
@@ -2419,6 +2460,126 @@ def _read_group_resolve_many2x_fields(self, data, fields):
+                tags = {tag[0]: tag for tag in tags}
+                for d in data:
+                    # use the "in" for the domain and not "="
+                    d['__domain'] = expression.AND([[(fullname, 'in', d[fullname])], domain])
+                    # replace tag raw value with list of raw value, label and color
+                    d[fullname] = tags.get(d[fullname])
+
+            elif property_type == 'many2many':
+                comodel = definition.get('comodel')
+                for d in data:
+                    # use the "in" for the domain and not "="
+                    d['__domain'] = expression.AND([[(fullname, 'in', d[fullname])], domain])
+                    # replace the id of the record by its lazy name get
+                    d[fullname] = lazy_name_get(self.env[comodel].browse(d[fullname]))[0]
+
+            elif property_type in ('date', 'datetime'):
+                for d in data:
+                    group_name = next((key for key in d.keys() if key.startswith(f"{fullname}:")), None)
+                    if not group_name or not d[group_name]:
+                        d[group_name] = False
+                        d['__range'] = {}
+                        continue
+                    fullname, func = group_name.split(':')
+                    display_formats = {
+                        'hour': 'hh:00 dd MMM',
+                        'day': 'dd MMM yyyy',
+                        'week': "'W'w YYYY",
+                        'month': 'MMMM yyyy',
+                        'quarter': 'QQQ yyyy',
+                        'year': 'yyyy',
+                    }
+                    assert func in display_formats
+
+                    # Date / Datetime are not JSONifiable, so they are stored as raw text
+                    db_format = '%Y-%m-%d' if property_type == 'date' else '%Y-%m-%d %H:%M:%S'
+
+                    if func != "week":
+                        start = (date_utils.start_of(d[group_name], func)).strftime(db_format)
+                        end = (date_utils.end_of(d[group_name], func) + datetime.timedelta(minutes=1)).strftime(db_format)
+                    else:
+                        # the value is the first day of the week (based on local)
+                        start = d[group_name].strftime(db_format)
+                        end = (d[group_name] + datetime.timedelta(days=7)).strftime(db_format)
+
+                    d.update({
+                        '__domain': expression.AND([
+                            [(fullname, '>=', start), (fullname, '<', end)],
+                            domain,
+                        ]),
+                        '__range': {group_name: {'from': start, 'to': end}},
+                        group_name: babel.dates.format_date(
+                            d[group_name],
+                            format=display_formats[func],
+                            locale=get_lang(self.env).code
+                        ),
+                    })
```

Comment:
Is there a specific reason why you are using the update function instead of modifying directly the values, like you did for the other types ? Other than to update `__domain`, `__range` and the value associated to the groupby at the same time ?